src/virtual_elements.js

```diff
@@ -275,13 +275,14 @@ var elementVoid = function(tag, key, statics, var_args) {
  * @param {...(function(string|number|boolean):string|number|boolean)} var_args
  *     Functions to format the value which are called only when the value has
  *     changed.
+ * @return {!Text} The corresponding Text Node.
  */
 var text = function(value, var_args) {
   if (process.env.NODE_ENV !== 'production') {
     assertNotInAttributes();
   }
 
-  var node = alignWithDOM('#text', null);
+  var node = /** @type {!Text}*/(alignWithDOM('#text', null));
```

Comment:
So I have just noticed I'm getting pull requests mixed up with #112. So will remove that comment for now.

As an aside the gslint does not parse the jsdoc types properly. Thats why you get lots of errors/warnings when you try and use the closure compiler to build your app.

This will cause issues for people that are trying to use idom in bigger codebases, like the closure library for example, which do use type annotations correctly.